theano/sandbox/mkl/__init__.py

```
@@ -0,0 +1,254 @@
+                                theano.Mode(optimizer=None),
+                                profile=False)
+            mkl_version.v = f()
+        except:
+            mkl_version.v = 0
+    return mkl_version.v
+
+mkl_version.v = None
+
+
+def _dnn_is_able_to_link():
+    if _dnn_is_able_to_link.status is None:
+        preambule = """
+            #include <stdio.h>
+            #include <mkl_dnn.h>
+            """
+        preambule += textwrap.dedent(header_text())
+
+        body = textwrap.dedent(
+            """
+            dnnError_t err;
+            dnnLayout_t usr_layout = NULL;
+            size_t size[1] = {256};
+            size_t stride[1] = {1};
+
+            if ((err = dnnLayoutCreate_F32(&usr_layout, 1, size, stride)) != E_SUCCESS) {
+                fprintf(stderr, "Failed to create user layout with mkl: %s", err);
+                return (-1);
+            }
+            """)
+
+        if 'mklml_intel' in config.blas.ldflags:
+            params = ['-l', 'mklml_intel', '-l', 'iomp5']
+        elif 'mklml_gnu' in config.blas.ldflags:
+            params = ['-l', 'mklml_gnu', '-l', 'iomp5']
+        elif 'mkl_rt' in config.blas.ldflags:
+            params = ['-l', 'mkl_rt']
+        else:
+            params = []
+
+        compilation_ok, run_ok, out, err = Compiler._try_flags(
+            flag_list=params, preambule=preambule, body=body,
+            try_run=True, output=True, compiler=theano.config.cxx,
+            comp_args=False)
+
+        mkl_available.msg = (
+            "Can not compile with MKL. We got this error: " +
+            str(err))
+        _dnn_is_able_to_link.status = run_ok
+
+    return _dnn_is_able_to_link.status
+
+_dnn_is_able_to_link.status = None
+
+
+def _dnn_is_enabled_by_user():
+    if config.device != "cpu":
+        mkl_available.msg = "MKL is disabled since device is not CPU. " \
+                            "Set the device to 'CPU' if you want to use MKL."
+        return False
```

Sorry, my first comment should have been:

If config.mkl.nn.enabled is True, you should raise an error.

(True instead of False)